src/domain/wallets/index.types.d.ts

```diff
@@ -179,20 +179,30 @@ type PaymentInputValidatorConfig = (
   walletId: WalletId,
 ) => Promise<Wallet | RepositoryError>
 
-type ValidatePaymentInputArgs = {
+type ValidatePaymentInputSenderArgs = {
   amount: number
   senderWalletId: string
   senderAccount: Account
-  recipientWalletId?: string
 }
+
 type ValidatePaymentInputRet = {
   amount: Satoshis
   senderWallet: Wallet
-  recipientWallet?: Wallet
+}
+
+type ValidatePaymentInputRecipientArgs = {
+  recipientWalletId: string
+  senderWallet: Wallet
+}
+type ValidatePaymentInputRecipientRet = {
```

Comment:
Minor nit, we've so far been using the `Result` suffix in places like this. Can probably standardise on one or the other

Reply:
@bodymindarts

Reply:
result: 26 occurences
ret: 6 occurences

Reply:
I like ret thought. few characters means fewer line return.

Reply:
I prefer Ret for return type. I associate Result with the ‘Result<T,E>’ type (which we aren’t using but is common in many languages)